core/import_data/import_records_section_C.py

```diff
@@ -0,0 +1,196 @@
+        return
+
+    current_country = {
+        "name": None,
+        "obj": None,
+    }
+    for index_row, row in df.iterrows():
+        # another country => another country program
+        if row["Country"] != current_country["name"]:
+            current_country["name"] = row["Country"]
+            current_country["obj"] = get_country(
+                current_country["name"], index_row, logger
+            )
+
+        # we didn't found a country in db:
+        if not current_country["obj"]:
+            continue
+
+        chemical_name = row["Controlled Substances"]
+
+        # I dont know what to do with this.
+        if chemical_name == "HFC-365mfc (93%)/HFC-227ea (7%) - mezcla":
+            continue
+
+        substance, blend = get_chemical(chemical_name, index_row, logger)
+        if not substance and not blend:
+            continue
+
+        previous_year_prices_obj = None
+        for report_details in REPORT_COLUMNS:
+            cp_report = get_cp_report(
+                report_details["year"],
+                current_country["obj"].name,
+                current_country["obj"].id,
+            )
+
+            try:
+                previous_year_price = (
+                    float(row[report_details["previous"]])
+                    if row[report_details["previous"]]
+                    else None
+                )
+                # some years are digits and some are strings
+                current_year = report_details["year"]
+                try:
+                    current_year_text = row[current_year]
+                    current_year_price = (
+                        float(current_year_text) if current_year_text else None
+                    )
+                except KeyError:
+                    current_year_text = row[str(current_year)]
+                    current_year_price = (
+                        float(current_year_text) if current_year_text else None
+                    )
+            # some price values are not decimals. skip them for now
+            except ValueError:
+                logger.warning(
+                    f"⚠️ [row: {index_row + OFFSET}] Price value is not a number."
+                )
+
+            # try to complete some missing data from previous year report
+            if previous_year_price:
+                if previous_year_prices_obj:
+                    if previous_year_prices_obj.current_year_price:
+                        if (
+                            previous_year_prices_obj.current_year_price
+                            != previous_year_price
+                        ):
+                            logger.warning(
+                                f"⚠️  [row: {index_row + OFFSET}]"
+                                f"[country: {current_country['obj'].name}][substance: {chemical_name}]"
+                                f"[year: {report_details['year']}] Mismatch in price declaration."
+                            )
+                    else:
```

Comment:
ce se intampla daca anul anterior are pretul null si textul "12-13". Iar updatul de anul asta vine cu valoarea 20?
E ok daca vom avea obiecte care sa aibe astfel de diferente?

Reply:
> E ok daca vom avea obiecte care sa aibe astfel de diferente?

teoretic, nu. practic, da 😅

aceast warning foarte detaliat este tocmai ca sa logam aceste discrepante. momentan, nu prea avem de unde sa stim care este valoarea corecta in cazul unei diferente ca sa o putem folosi pe aceea.  din acest motiv, face update doar daca nu exista o valoarea declarata in raportarea anterioara.

> ce se intampla daca anul anterior are pretul null si textul "12-13". Iar updatul de anul asta vine cu valoarea 20?

am ales sa tratez doar cazul in care putem extrage un pret pentru ca `pretul null si textul "12-13"` este o anomalie si nu ar fi prea frumos sa adaugam inca una